src/common/geometry.js

```diff
@@ -31,3 +33,41 @@ export const slope = (v1, v2) => {
   let n = v2.y - v1.y
   return d === 0 ? undefined : n / d
 }
+
+// computes a bounding box from a set of points.
+export const findBounds = (vertices) => {
+  let n = vertices.length
+
+  if (n === 0) {
+    return []
+  }
+
+  let loX = vertices[0].x
+  let hiX = vertices[0].x
+  let loY = vertices[0].y
+  let hiY = vertices[0].y
+
+  for(let i=1; i<n; ++i) {
+    let p = vertices[i]
+    loX = Math.min(loX, p.x)
+    hiX = Math.max(hiX, p.x)
+    loY = Math.min(loY, p.y)
+    hiY = Math.max(hiY, p.y)
+  }
+
+  return [new Victor(loX, loY), new Victor(hiX, hiY)]
+}
+
+// resizes each vertex in a list to the specified dimensions. Will not stretch the shape.
+export const resizeVertices = (vertices, sizeX, sizeY) => {
```

Comment:
A lot of this looks very similar to the stuff I just finished to stretch the gcode to fill the space.

Reply:
Should we attempt to reconcile afterward, assuming this gets merged?